providers/ms365/resources/identity_and_access.go

```diff
@@ -0,0 +1,188 @@
+	}
+
+	var policyResources []interface{}
+	if policies == nil {
+		return nil, nil
+	}
+
+	for _, policy := range policies.GetValue() {
+		if policy.GetId() != nil && policy.GetDisplayName() != nil {
+			policyResource, err := newMqlRoleManagementPolicy(a.MqlRuntime, policy)
+			if err != nil {
+				return nil, fmt.Errorf("failed to create MQL resource for policy ID %s: %w", *policy.GetId(), err)
+			}
+			policyResources = append(policyResources, policyResource)
+		}
+	}
+
+	return policyResources, nil
+}
+
+func newMqlRoleManagementPolicy(runtime *plugin.Runtime, u models.UnifiedRoleManagementPolicyable) (*mqlMicrosoftIdentityAndAccessPolicy, error) {
+	lastModifiedByDict := map[string]interface{}{}
+
+	if u.GetLastModifiedBy() != nil {
+		lastModifiedByDict["id"] = llx.StringDataPtr(u.GetLastModifiedBy().GetId())
+		lastModifiedByDict["displayName"] = llx.StringDataPtr(u.GetLastModifiedBy().GetDisplayName())
+	}
+
+	resource, err := CreateResource(runtime, "microsoft.identityAndAccess.policy",
+		map[string]*llx.RawData{
+			"__id":                  llx.StringDataPtr(u.GetId()),
+			"id":                    llx.StringDataPtr(u.GetId()),
+			"displayName":           llx.StringDataPtr(u.GetDisplayName()),
+			"description":           llx.StringDataPtr(u.GetDescription()),
+			"isOrganizationDefault": llx.BoolDataPtr(u.GetIsOrganizationDefault()),
+			"scopeId":               llx.StringDataPtr(u.GetScopeId()),
+			"scopeType":             llx.StringDataPtr(u.GetScopeType()),
+			"lastModifiedDateTime":  llx.TimeDataPtr(u.GetLastModifiedDateTime()),
+			"lastModifiedBy":        llx.DictData(lastModifiedByDict),
+		})
+	if err != nil {
+		return nil, err
+	}
+	return resource.(*mqlMicrosoftIdentityAndAccessPolicy), nil
+}
+
+func (m *mqlMicrosoftIdentityAndAccessPolicy) rules() ([]interface{}, error) {
+	conn := m.MqlRuntime.Connection.(*connection.Ms365Connection)
+	graphClient, err := conn.GraphClient()
+	if err != nil {
+		return nil, err
+	}
+
+	policyId := m.Id.Data
+	if policyId == "" {
+		return nil, fmt.Errorf("policy resource has an empty ID, cannot fetch rules")
+	}
+
+	ctx := context.Background()
+
+	rulesResult, err := graphClient.Policies().RoleManagementPolicies().ByUnifiedRoleManagementPolicyId(policyId).Rules().Get(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get rules for policy %s: %w", policyId, err)
+	}
+
+	var ruleResources []interface{}
+	if rulesResult == nil {
+		return nil, nil
+	}
+
+	for _, rule := range rulesResult.GetValue() {
+		if rule.GetId() == nil {
+			continue
+		}
+		ruleResource, err := newMqlRoleManagementPolicyRule(m.MqlRuntime, rule)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create MQL resource for rule ID %s: %w", *rule.GetId(), err)
+		}
+		ruleResources = append(ruleResources, ruleResource)
+	}
+
+	return ruleResources, nil
+}
+
+func newMqlRoleManagementPolicyRule(runtime *plugin.Runtime, rule models.UnifiedRoleManagementPolicyRuleable) (plugin.Resource, error) {
```

Comment:
Shouldn't this return a specific type instead of `plugin.Resource`?

Like, e.g., here: https://github.com/mondoohq/cnquery/blob/0a71a15ef690bb82e9b81056c277ffcaa287bf81/providers/ms365/resources/devices.go#L225